Configurations/10-main.conf

```diff
@@ -1375,6 +1375,12 @@ sub vms_info {
         perlasm_scheme   => "auto",
         multilib         => "-x64",
     },
+    "VC-WINCORE64A" => {
+        inherit_from     => [ "VC-WIN64A" ],
+        cflags           => add("-DOPENSSL_SYS_WIN_CORE  -DOPENSSL_NO_CAPIENG -DOPENSSL_NO_ASYNC"),
+        lflags           => add("/NODEFAULTLIB:kernel32.lib /NODEFAULTLIB:ws2_32.lib /NODEFAULTLIB:gdi32.lib /NODEFAULTLIB:advapi32.lib /NODEFAULTLIB:crypt32.lib /NODEFAULTLIB:user32.lib /NXCOMPAT /DYNAMICBASE /machine:X64"),
+        ex_libs          => "onecore.lib",
```

Comment:
I would actually suggest to link with onecore_downlevel. We are still talking about desktop OSes here, so downlevel would be appropriate and safe bet. I would guess that none of extra definitions above would be required then. As for non-desktop OSes [such as IoT], *they* should be denoted by additional defines [in case they can't be identified by a pre-define, which would be preferred option.]

Reply:
OPENSSL_SYS_WIN_CORE definition here was designed to disable unsupported by OneCoreAPI methods and also these unsupported by IoT, like MessageBoxes etc. Idea was to provide selective compilation which can run on literally every modern and supported by MS Windows.
I see it disables some functionalities on desktop Windows which is unnecessary.
As mentioned somewhere else, I wonder leaving some method calls directly linked to system libs, makes Openssl not OneCore capable, so some definition for OneCore probably will be needed.
Additionally some features can not run on non-desktop OS'es (which I disabled using that define, and they seemed to be unnecessary). Despite of whether We totally migrate legacy LoadLibrary to LoadLibraryEx etc functions or not, there still is usage of e.g MessageBox. I agree its good idea to possibly use separate define, and for IoT just build Core configuration with additional one.